backend/app/services/scenario_preparation_service.py

```diff
@@ -18,24 +18,33 @@
     ScenarioPreparationRequest,
     StringListResponse,
 )
+from app.services.vector_db_context_service import query_vector_db_and_prompt
 
 
 @retry(stop=stop_after_attempt(3), wait=wait_fixed(1))
```

Comment:
Why do we retry 3 times and why do we wait between the retries?
Is it because the API call might fail or is it because the GenAI output may be misformed and cant be parsed?